docpages/building/linux.md

```diff
@@ -37,27 +37,29 @@ Once installed to the `/usr/local` directory, you can make use of the library in
 g++ -std=c++17 mydppbot.cpp -o dppbot -ldpp
 ```
 
-If you are on **Arch Linux**, you will need to add `/usr/local` to your environment variables:
+If you are on **Arch Linux**:
 
+- You need to give proper permission to `libdpp.so` and `libdpp.so.10.0.29`
 ```bash
-export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
-export CPLUS_INCLUDE_PATH=/usr/local/include:$CPLUS_INCLUDE_PATH
+sudo chmod 755 /usr/local/lib/libdpp.so.10.0.29
 ```
 
-However, if you don't want to change your environment settings, you can use these flags to compile:
-
+- You need to specify the location for `libdpp.so` when compilling:
 ```bash
-g++ -std=c++17 -L/usr/local/lib -I/usr/local/include mydppbot.cpp -o dppbot -ldpp
+g++ -std=c++17 -I/usr/local/include mydppbot.cpp -o dppbot /usr/local/lib/libdpp.so -Wl,-rpath,/usr/local/lib
 ```
 
 The important flags in this command-line are:
 
 * `-std=c++17` - Required to compile the headers
-* `-L/usr/local/lib` - Required to tell the linker where libdpp is located.
-* `-I/usr/local/include` - Required to tell the linker where dpp headers are located.
+* `-L/usr/local/lib` - Required to tell the linker where libdpp is located
```

Comment:
-L isn't used here anymore.

Reply:
also `-ldpp` but i fixed it

Reply:
Why is `-ldpp` not required anymore? Is it because of `/usr/local/lib/libdpp.so`? If so, can you see about swapping it for `-ldpp`.

Reply:
`-lpdd` is cant not be used because it will tell linker to find it in standard location which is `/usr/lib` thats why.

`/usr/local/lib/libdpp.so` adding this specify the path of library
and then
`-Wl,-rpath,/usr/local/lib` adding this specigy the path of directory
it is required for some werid reason i dont know, its not running with that, if we sont specify the path of directory then it wont work, we need to specify the library file as well as directory

Reply:
> But if the CMake original install location is `/usr/local/lib`, then `-ldpp` should look there?

no thats the problem, cmake install it in `/usr/local/lib` and `-ldpp` is set to standard location, which in other linux(s) the standard location is `/usr/local/lib` but in arch it is `/usr/lib`, cmake install it in the **specific** place and then that **specific** place is standard for debian and all not for arch

Reply:
how strange, alrighty.